skbio/io/format/embl.py

```diff
@@ -0,0 +1,1568 @@
+
+
+# Writer methods
+@embl.writer(None)
+def _generator_to_embl(obj, fh):
+    for obj_i in obj:
+        _serialize_single_embl(obj_i, fh)
+
+
+@embl.writer(Sequence)
+def _sequence_to_embl(obj, fh):
+    _serialize_single_embl(obj, fh)
+
+
+@embl.writer(DNA)
+def _dna_to_embl(obj, fh):
+    _serialize_single_embl(obj, fh)
+
+
+@embl.writer(RNA)
+def _rna_to_embl(obj, fh):
+    _serialize_single_embl(obj, fh)
+
+
+@embl.writer(Protein)
+def _protein_to_embl(obj, fh):
+    # no protein support, at the moment
+    raise EMBLFormatError("There's no protein support for EMBL record. "
+                          "Current status of EMBL protein support is "
+                          "described in issue-1499 (https://github.com/"
+                          "biocore/scikit-bio/issues/1499)")
+
+
+def _construct(record, constructor=None, **kwargs):
+    '''Construct the object of Sequence, DNA, RNA, or Protein.'''
+
+    # sequence, metadata and interval metadata
+    seq, md, imd = record
+
+    if 'lowercase' not in kwargs:
+        kwargs['lowercase'] = True
+
+    if constructor is None:
+        unit = md['LOCUS']['unit']
+        if unit == 'bp':
+            # RNA mol type has T instead of U for genbank from from NCBI
+            constructor = DNA
+
+        elif unit == 'aa':
+            # no protein support, at the moment
+            # constructor = Protein
+            raise EMBLFormatError("There's no protein support for EMBL record")
+
+    if constructor == RNA:
+        return DNA(
+            seq, metadata=md, interval_metadata=imd, **kwargs).transcribe()
+    else:
+        return constructor(
+            seq, metadata=md, interval_metadata=imd, **kwargs)
+
+
+# looks like the genbank _parse_genbank
+def _parse_embls(fh):
+    """Chunck multiple EMBL records by '//', and returns a generator"""
+
+    data_chunks = []
+    for line in _line_generator(fh, skip_blanks=True, strip=False):
+        if line.startswith('//'):
+            yield _parse_single_embl(data_chunks)
+            data_chunks = []
+        else:
+            data_chunks.append(line)
+
+
+def _parse_single_embl(chunks):
+    metadata = {}
+    interval_metadata = None
+    sequence = ''
+
+    # define a section splitter with _embl_yield_section function defined in
+    # this module (return the embl section by embl key). returns generator for
+    # each block with different line type
+    section_splitter = _embl_yield_section(
+        lambda line: _get_embl_section(line),
+        skip_blanks=True,
+        strip=False)
+
+    # process each section, like genbank does
+    for section in section_splitter(chunks):
+        # embl_key is line type (eg ID, AC, ...)
+        embl_key = _get_embl_key(section[0])
+
+        # even section have different keys, (RA, RP, ...), I need to get the
+        # correct section to call the appropriate method (eg REFERENCE)
+        embl_section = _get_embl_section(embl_key)
```

Comment:
changed `_embl_yield_section`. I've renamed `embl_section` in `section_name`, it seems clearer to me